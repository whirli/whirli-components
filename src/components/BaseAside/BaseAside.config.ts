// @ts-ignore
import { PropType } from '@composition';
import * as PropHelpers from '@whirli-components/helpers/props';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseAside/BaseAside.config';

// Package config
import { PropKeys, PropValues } from './BaseAside.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
