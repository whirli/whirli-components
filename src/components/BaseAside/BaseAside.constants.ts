export const PropKeys = {};

export const PropValues = {};
