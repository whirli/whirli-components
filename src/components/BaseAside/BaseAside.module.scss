@import '../../styles/Breakpoints.scss';

// default styles
.aside {}

// breakpoint styles
$packageBreakpointStyles: ();