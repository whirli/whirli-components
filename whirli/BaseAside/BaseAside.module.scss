@import '@whirli-components/src/components/BaseAside/BaseAside.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));