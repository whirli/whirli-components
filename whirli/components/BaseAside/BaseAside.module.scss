@import '@whirli-components/components/BaseAside/BaseAside.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));