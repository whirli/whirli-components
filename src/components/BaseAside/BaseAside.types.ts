import { PropKeys, PropValues } from './BaseAside.constants';
