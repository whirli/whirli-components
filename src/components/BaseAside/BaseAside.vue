<template>
  <aside id="aside" :class="classes">
    <slot />
  </aside>
</template>

<script setup lang="ts">
// Vue
// Styles
import styles from '@whirli-local/components/BaseAside/BaseAside.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseAside.config';

const ComponentStyles = ConfigStyles;

const props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes = [styles.aside, ...makeClasses(ComponentStyles, props, styles)];
</script>
