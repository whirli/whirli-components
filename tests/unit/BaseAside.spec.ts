import { shallowMount } from '@vue/test-utils';
import BaseAside from '@/components/BaseAside/BaseAside.vue';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const text = 'Hello World';

describe('BaseAside.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseAside, {
      slots: {
        default: text,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.text()).toMatch(text);
  });
});
