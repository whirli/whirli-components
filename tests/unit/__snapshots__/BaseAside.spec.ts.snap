// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseAside.vue - Single classes renders the component 1`] = `
<aside
  class=""
  id="aside"
>
  
  Hello World
  
</aside>
`;
